src/Ocelot/Metadata/DownstreamRouteExtensions.cs

```diff
@@ -0,0 +1,164 @@
+    /// Some custom conversion has been for the following types:
+    /// bool, bool?, string[], numeric types
+    /// otherwise trying to deserialize the value using the JsonSerializer
+    /// </summary>
+    /// <param name="targetType">The target type.</param>
+    /// <param name="value">The string value.</param>
+    /// <param name="metadataOptions">The metadata options, it includes the global configuration.</param>
+    /// <param name="jsonSerializerOptions">If needed, some custom json serializer options.</param>
+    /// <returns>The converted string.</returns>
+    private static object ConvertTo(Type targetType, string value, MetadataOptions metadataOptions,
+        JsonSerializerOptions jsonSerializerOptions)
+    {
+        if (targetType == typeof(string))
+        {
+            return value;
+        }
+
+        if (targetType == typeof(bool))
+        {
+            return TruthyValues.Contains(value.Trim());
+        }
+
+        if (targetType == typeof(bool?))
+        {
+            if (TruthyValues.Contains(value.Trim()))
+            {
+                return true;
+            }
+
+            if (FalsyValues.Contains(value.Trim()))
+            {
+                return false;
+            }
+
+            return null;
+        }
+
+        if (targetType == typeof(string[]))
+        {
+            if (value == null)
+            {
+                return Array.Empty<string>();
+            }
+
+            return value.Split(metadataOptions.Separators, metadataOptions.StringSplitOption)
+                .Select(s => s.Trim(metadataOptions.TrimChars))
+                .Where(s => !string.IsNullOrEmpty(s))
+                .ToArray();
+        }
+
+        return NumericTypes.Contains(targetType)
+            ? ConvertToNumericType(value, targetType, metadataOptions.CurrentCulture, metadataOptions.NumberStyle)
+            : JsonSerializer.Deserialize(value, targetType, jsonSerializerOptions);
+    }
+
+    /// <summary>
+    /// Using reflection to invoke the Parse method of the numeric type
+    /// </summary>
+    /// <param name="value">The number as string.</param>
+    /// <param name="targetType">The target numeric type.</param>
+    /// <param name="provider">The current format provider.</param>
+    /// <param name="numberStyle">The current number style configuration.</param>
+    /// <returns>The parsed string as object of type targetType.</returns>
+    /// <exception cref="InvalidOperationException">Exception thrown if the type doesn't contain a "Parse" method. This shouldn't happen.</exception>
+    private static object ConvertToNumericType(string value, Type targetType, IFormatProvider provider,
+        NumberStyles numberStyle)
+    {
+        MethodInfo parseMethod =
+            targetType.GetMethod("Parse", new[] { typeof(string), typeof(NumberStyles), typeof(IFormatProvider) }) ??
+            throw new InvalidOperationException("No suitable parse method found.");
```

Comment:
### Oh my gosh! 🤯 

### Reflection Code in the lib❗
Employing reflection in production code is detrimental as it impairs performance.
While reflection reduces the amount of code, it comes at the cost of diminished performance.

Eliminate the use of reflection and incorporate explicit type analysis statements. The `IConvertible` interfaces are likely to aid in reworking the analysis process.

Reply:
Ok, I prefer a switch on all numeric types. It's the fastest method. More code, but fast.